src/Functions/translate.cpp

```diff
@@ -361,9 +363,18 @@ class FunctionTranslate : public IFunction
         String map_from = c1_const->getValue<String>();
         String map_to = c2_const->getValue<String>();
 
-        auto map_from_size = UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(map_from.data()), map_from.size());
-        auto map_to_size = UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(map_to.data()), map_to.size());
-
+        size_t map_from_size;
+        size_t map_to_size;
+        if constexpr (std::is_same_v<Impl, TranslateUTF8Impl>)
+        {
+            map_from_size = UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(map_from.data()), map_from.size());
+            map_to_size = UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(map_to.data()), map_to.size());
+        }
+        else
+        {
+            map_from_size = map_from.size();
+            map_to_size = map_to.size();
+        }
```

Comment:
Maybe you want to use `std::string::length`? AFAIU, this method is designed to keep in mind knowledge about UTF-8.

Reply:
NVM, it's not :(